File: PeerInfo.java
import java.io.*;
import java.util.*;

public class PeerInfo {
    int peerId;
    String hostName;
    int port;
    boolean hasFile;

    public PeerInfo(int peerId, String hostName, int port, boolean hasFile) {
        this.peerId = peerId;
        this.hostName = hostName;
        this.port = port;
        this.hasFile = hasFile;
    }

    public static List<PeerInfo> loadAll(String path) {
        List<PeerInfo> list = new ArrayList<>();
        try (BufferedReader br = new BufferedReader(new FileReader(path))) {
            String line;
            while ((line = br.readLine()) != null) {
                line = line.trim();
                if (line.isEmpty()) continue;
                String[] parts = line.split("\\s+");
                int id = Integer.parseInt(parts[0]);
                String host = parts[1];
                int port = Integer.parseInt(parts[2]);
                boolean hasFile = parts[3].trim().equals("1");
                list.add(new PeerInfo(id, host, port, hasFile));
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
        return list;
    }
}

File: peer_info.py
class PeerInfoEntry:
    def __init__(self, peer_id, host_name, port, has_file):
        self.peer_id = peer_id
        self.host_name = host_name
        self.port = port
        self.has_file = has_file

    @staticmethod
    def load_all(path):
        peers = []
        with open(path, "r") as f:
            for line in f:
                line = line.strip()
                if not line:
                    continue
                parts = line.split()
                peer_id = int(parts[0])
                host_name = parts[1]
                port = int(parts[2])
                has_file = parts[3].strip() == "1"
                peers.append(PeerInfoEntry(peer_id, host_name, port, has_file))
        return peers
